### PR TITLE
[ENHANCEMENT] When connecting to new Datasource, CLI prompt is consistent with rest of GE

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Changes proposed in this pull request:
- When a user connects to an SQL database as a new datasource, and chooses to list all the tables in the DB, the prompt now asks `Would you like to proceed? [Y/n]:`, which is consistent with the rest of the CLI.  

```
You have selected a datasource that is a SQL database. How would you like to specify the data?
1. Enter a table name and schema
2. Enter a custom SQL query
3. List all tables in the database (this may take a very long time)
: 3
Warning: If you have a large number of tables in your datasource, this may take a very long time.
Would you like to proceed? [Y/n]:

```
- There were also some strange characters that had been introduced in the output (like `r` and `\m`). These have been cleaned up)

### CODE REVIEW

File: great_expectations/cli/datasource.py [hunk 1]
```diff
@@ -1282,12 +1282,9 @@ def _get_batch_kwargs_for_sqlalchemy_datasource(
...
 
         elif single_or_multiple_data_asset_selection == "3":  # list it all
-            msg_prompt_warning = fr"""Warning: If you have a large number of tables in your datasource, this may take a very long time. \m
-                    Would you like to continue?"""
-            confirmation = click.prompt(
-                msg_prompt_warning, type=click.Choice(["y", "n"]), show_choices=True
-            )
-            if confirmation == "y":
+            msg_prompt_warning = f"""Warning: If you have a large number of tables in your datasource, this may take a very long time.\nWould you like to proceed?"""
+            confirmation = click.confirm(msg_prompt_warning, default=True)
```
Comment (Shinnnyshinshin) [inline great_expectations/cli/datasource.py:1286]: `click.confirm()` is what is used by commands in the CLI that start a potentially lengthy (>1 sec) computation or modifies some resources (e.g., edits the config file, adds objects to the stores).  The method allows the user to confirm the command or exits the program.

In our case, we don't actually want to exit GE, so we just keep the same prompt.

Comment (spbail) [reply great_expectations/cli/datasource.py:1286]: Does this print the default [Y/n]?

Comment (Shinnnyshinshin) [reply great_expectations/cli/datasource.py:1286]: yes. `click.confirm()` is what prints the `[Y/n]` part. The `msg_prompt_warning` is everything leading up to it including `Would you like to proceed?`

Comment (spbail) [reply great_expectations/cli/datasource.py:1286]: Nice!

**Review by spbail — APPROVED** [review]
lgtm, thanks!

**Pull request merged**